logging/logrus/context.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2016 Michal Witkowski. All Rights Reserved.
+// See LICENSE for licensing terms.
+
+package http_logrus
+
+import (
+	"net/http"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/mwitkow/go-httpwares/tags"
+	"golang.org/x/net/context"
+)
+
+type ctxMarker struct{}
+
+var (
+	ctxMarkerKey = &ctxMarker{}
+)
+
+// Extract takes the call-scoped logrus.Entry from grpc_logrus middleware.
+//
+// The logger will have fields pre-populated using http_ctxtags.
+//
+// If the http_logrus middleware wasn't used, a no-op `logrus.Entry` is returned. This makes it safe to use regardless.
+func Extract(req *http.Request) *logrus.Entry {
+	return ExtractFromContext(req.Context())
+}
+
+// Extract takes the call-scoped logrus.Entry from grpc_logrus middleware.
+//
+// The logger will have fields pre-populated using http_ctxtags.
+//
+// If the http_logrus middleware wasn't used, a no-op `logrus.Entry` is returned. This makes it safe to use regardless.
+func ExtractFromContext(ctx context.Context) *logrus.Entry {
+	l, ok := ctx.Value(ctxMarkerKey).(*logrus.Entry)
+	if !ok {
+		return logrus.NewEntry(nullLogger)
```

Comment:
You would magically lose logs when you do this without any warning, is it not worth just using a standard out logger in case things go wrong?